src/main/java/net/hypixel/nerdbot/curator/ForumChannelCurator.java

```diff
@@ -0,0 +1,92 @@
+            log("Curating forum channel: " + forumChannel.getName() + " (Channel ID: " + forumChannel.getId() + ")");
+
+            for (ThreadChannel thread : forumChannel.getThreadChannels()) {
+                log("Curating thread: " + thread.getName() + " (Thread ID: " + thread.getId() + ")");
+
+                // Check if the post is already greenlit then skip it
+                if (thread.getAppliedTags().stream().anyMatch(tag -> tag.getName().equalsIgnoreCase("greenlit"))) {
+                    log("Thread '" + thread.getName() + "' is already greenlit, skipping...");
+                    continue;
+                }
+
+                try {
+                    // This is a stupid way to do it but it's the only way that works right now
+                    List<Message> allMessages = thread.getIterableHistory().complete(true);
+                    Message firstPost = allMessages.get(allMessages.size() - 1);
+                    Emoji agreeEmoji = getBot().getJDA().getEmojiById(getBot().getConfig().getEmojis().getAgree());
+                    Emoji disagreeEmoji = getBot().getJDA().getEmojiById(getBot().getConfig().getEmojis().getDisagree());
+
+                    if (agreeEmoji == null || disagreeEmoji == null) {
+                        log("Couldn't find the agree or disagree emoji, time to yell!");
+                        ChannelManager.getLogChannel().sendMessage(Users.getUser(Users.AERH).getAsMention() + " I couldn't find the agree or disagree emoji for some reason, check logs!").queue();
+                        break;
+                    }
+
+                    // Add the reactions if they're missing
+                    // Theoretically the agree reaction should be there by default but lets check anyway just in case
+                    MessageReaction agreeReaction = firstPost.getReaction(agreeEmoji);
+                    MessageReaction disagreeReaction = firstPost.getReaction(disagreeEmoji);
+
+                    if (agreeReaction == null) {
+                        log("Adding a missing agree reaction to post " + firstPost.getId());
+                        firstPost.addReaction(agreeEmoji).complete();
+                    }
+
+                    if (disagreeReaction == null) {
+                        log("Adding a missing disagree reaction to post " + firstPost.getId());
+                        firstPost.addReaction(disagreeEmoji).complete();
+                    }
+
+                    if (agreeReaction.getCount() < getBot().getConfig().getMinimumThreshold()) {
+                        log("Post " + firstPost.getId() + " doesn't have enough agree reactions, skipping...");
+                        continue;
+                    }
+
+                    // Get the ratio of reactions and greenlight it if it's over the threshold
+                    double ratio = getRatio(agreeReaction.getCount(), disagreeReaction.getCount());
+                    if (ratio >= getBot().getConfig().getPercentage()) {
```

Comment:
check to make sure percentage is set in config

Reply:
It'll default to 0 if it's not set in the config but would be a good idea to set default values